### DISCUSSION TITLE
CI/CD for the rewrite

### DISCUSSION THREAD

**Discussion by oskardotglobal** [winapps-org/winapps#22] [Ideas]
Currently, [the checks on the rewrite branch are failing](https://github.com/winapps-org/winapps/actions/runs/5562882496/job/15067664362) because rustfmt wants formatting changes. We should look into preventing this in the future by using tools like https://pre-commit.com to prevent stuff like this in the future.

**Comment by LDprg** [comment]
Yeah, we should definitely do this (I currently have no time actually code anything).

**Comment by oskardotglobal** [comment]
I've added a pre-commit config along with basic usage guidelines to the rewrite branch and am currently figuring out how to get pre-commit ci working with a non-default branch

**Reply by LDprg** [reply]
@oskardotglobal How about additional  autoformatting on commit (to keep the code always formatted)? For this we could use the official rust formatter (cargo fmt), I am pretty sure that isnt able to break code and only formats it.

The only thing is that I am no sure how to realised such a thing with Github Actions. As you need to change the code and make a second commit or pr. Is pre-commit only a checking tool or does it allow automatic changes in the code?

**Reply by oskardotglobal** [reply]
Pre-commit runs cargo fmt and also automatically fixes PRs

**Reply by oskardotglobal** [reply]
I'm currently writing the rust wrapper for quickemu and will open a PR with those changes soon, then we'll see how it works
